go-controller/pkg/cluster/master.go

```diff
@@ -224,9 +224,13 @@ func (cluster *OvnClusterController) StartClusterMaster(masterNodeName string) e
 	// but omitting any that exist in 'subrange' (third argument)
 	for _, clusterEntry := range cluster.ClusterIPNet {
 		subrange := make([]string, 0)
-		for _, entry := range alreadyAllocated {
-			if clusterEntry.CIDR.Contains(net.ParseIP(entry)) {
-				subrange = append(subrange, entry)
+		for _, allocatedRange := range alreadyAllocated {
+			firstAddress, _, err := net.ParseCIDR(allocatedRange)
+			if err != nil {
+				return err
+			}
+			if clusterEntry.CIDR.Contains(firstAddress) {
```

Comment:
looks good. Will mess the logic a bit if there were overlapping addresses in ClusterIPNet. I hope we can put in a validator for that somewhere (another PR).